### ISSUE TITLE
Something seems seems wrong - npm run dev 

### ISSUE THREAD

**Issue by HaroldMitts** [microsoft/JARVIS#79]
I can run in the terminal, 
![image](https://user-images.githubusercontent.com/13913671/230505160-545f2ec5-810d-4531-9df2-2a0cd5571f41.png)

but something is wrong with the web interface and I cannot figure out what. What can I do to troubleshoot?

![image](https://user-images.githubusercontent.com/13913671/230505122-acdf4e6f-a97c-4bc4-8dcd-d1e4605eeb6c.png)

Here is the config.yaml and the lite.yaml is similar

```
openai:
  key: "sk-ke------------------------------" # "gradio" (set when request) or your_personal_key
huggingface:
  token: "hf_si-------------------------------" # required: huggingface token @ https://huggingface.co/settings/tokens
local: # ignore: just for development
  endpoint: http://localhost:8003
  
dev: false
debug: false
log_file: logs/debug.log
model: text-davinci-003 # currently only support text-davinci-003, we will support more open-source LLMs in the future
use_completion: true
inference_mode: huggingface # local, huggingface or hybrid, prefer hybrid
local_deployment: full # minimal, standard or full, prefer full
num_candidate_models: 5
max_description_length: 100
proxy: # optional: your proxy server "http://ip:port"
httpserver:
  host: localhost
  port: 8004
modelserver:
  host: localhost
  port: 8005
logit_bias:
  parse_task: 0.1
  choose_model: 5
tprompt:
  parse_task: >-
    #1 Task Planning Stage: The AI assistant can parse user input to several tasks: [{"task": task, "id": task_id, "dep": dependency_task_id, "args": {"text": text or <GENERATED>-dep_id, "image": image_url or <GENERATED>-dep_id, "audio": audio_url or <GENERATED>-dep_id}}]. The special tag "<GENERATED>-dep_id" refer to the one genereted text/image/audio in the dependency task (Please consider whether the dependency task generates resources of this type.) and "dep_id" must be in "dep" list. The "dep" field denotes the ids of the previous prerequisite tasks which generate a new resource that the current task relies on. The "args" field must in ["text", "image", "audio"], nothing else. The task MUST be selected from the following options: "token-classification", "text2text-generation", "summarization", "translation", "question-answering", "conversational", "text-generation", "sentence-similarity", "tabular-classification", "object-detection", "image-classification", "image-to-image", "image-to-text", "text-to-image", "text-to-video", "visual-question-answering", "document-question-answering", "image-segmentation", "depth-estimation", "text-to-speech", "automatic-speech-recognition", "audio-to-audio", "audio-classification", "canny-control", "hed-control", "mlsd-control", "normal-control", "openpose-control", "canny-text-to-image", "depth-text-to-image", "hed-text-to-image", "mlsd-text-to-image", "normal-text-to-image", "openpose-text-to-image", "seg-text-to-image". There may be multiple tasks of the same type. Think step by step about all the tasks needed to resolve the user's request. Parse out as few tasks as possible while ensuring that the user request can be resolved. Pay attention to the dependencies and order among tasks. If the user input can't be parsed, you need to reply empty JSON []. 
  choose_model: >-
    #2 Model Selection Stage: Given the user request and the parsed tasks, the AI assistant helps the user to select a suitable model from a list of models to process the user request. The assistant should focus more on the description of the model and find the model that has the most potential to solve requests and tasks. Also, prefer models with local inference endpoints for speed and stability.
  response_results: >-
    #4 Response Generation Stage: With the task execution logs, the AI assistant needs to describe the process and inference results.
demos_or_presteps:
  parse_task: demos/demo_parse_task.json
  choose_model: demos/demo_choose_model.json
  response_results: demos/demo_response_results.json 
prompt:
  parse_task: The chat log [ {{context}} ] may contain the resources I mentioned. Now I input { {{input}} }. Pay attention to the input and output types of tasks and the dependencies between tasks.
  choose_model: >-
    Please choose the most suitable model from {{metas}} for the task {{task}}. The output must be in a strict JSON format: {"id": "id", "reason": "your detail reasons for the choice"}.
  response_results: >-
    Yes. Please first think carefully and directly answer my request based on the inference results. Then please detail your workflow step by step including the used models and inference results for my request in your friendly tone. Please filter out information that is not relevant to my request. If any generated files of images, audios or videos in the inference results, must tell me the complete path. If there is nothing in the results, please tell me you can't make it. }
```

**Comment by e-oldman** [comment]
Might be the same issue - https://github.com/microsoft/JARVIS/issues/58

**Comment by HaroldMitts** [comment]
I noticed in the issue #58 the logs were requested. Here are my logs.

log_success.jsonl

```
{"input": "are you there?", "task": "[{\"task\": \"conversational\", \"id\": 0, \"dep\": [-1], \"args\": {\"text\": \"are you there?\" }}]", "results": {"0": {"task": {"task": "conversational", "id": 0, "dep": [-1], "args": {"text": "are you there?"}}, "inference result": {"response": "Yes, I am here. What can I help you with?"}, "choose model result": {"id": "ChatGPT", "reason": "ChatGPT is the best model for this task."}}}, "response": "Sure. I understand your request. Based on the inference results, my response to your request is \"Yes, I am here. What can I help you with?\". My workflow for this request is as follows: First, I received your request \"are you there?\" and identified it as a conversational task. Then, I chose the ChatGPT model as the best model for this task. Finally, I used the ChatGPT model to generate the response \"Yes, I am here. What can I help you with?\". There are no generated files of images, audios or videos in the inference results. Is there anything else I can help you with?", "during": 10.202335119247437, "op": "response"}
```

debug.log

```
2023-04-06 12:17:34,950 - __main__ - INFO - ********************************************************************************
2023-04-06 12:17:34,950 - __main__ - INFO - input: are you there?
2023-04-06 12:17:34,951 - __main__ - DEBUG - [{'role': 'system', 'content': '#1 Task Planning Stage: The AI assistant can parse user input to several tasks: [{"task": task, "id": task_id, "dep": dependency_task_id, "args": {"text": text or <GENERATED>-dep_id, "image": image_url or <GENERATED>-dep_id, "audio": audio_url or <GENERATED>-dep_id}}]. The special tag "<GENERATED>-dep_id" refer to the one genereted text/image/audio in the dependency task (Please consider whether the dependency task generates resources of this type.) and "dep_id" must be in "dep" list. The "dep" field denotes the ids of the previous prerequisite tasks which generate a new resource that the current task relies on. The "args" field must in ["text", "image", "audio"], nothing else. The task MUST be selected from the following options: "token-classification", "text2text-generation", "summarization", "translation", "question-answering", "conversational", "text-generation", "sentence-similarity", "tabular-classification", "object-detection", "image-classification", "image-to-image", "image-to-text", "text-to-image", "text-to-video", "visual-question-answering", "document-question-answering", "image-segmentation", "depth-estimation", "text-to-speech", "automatic-speech-recognition", "audio-to-audio", "audio-classification", "canny-control", "hed-control", "mlsd-control", "normal-control", "openpose-control", "canny-text-to-image", "depth-text-to-image", "hed-text-to-image", "mlsd-text-to-image", "normal-text-to-image", "openpose-text-to-image", "seg-text-to-image". There may be multiple tasks of the same type. Think step by step about all the tasks needed to resolve the user\'s request. Parse out as few tasks as possible while ensuring that the user request can be resolved. Pay attention to the dependencies and order among tasks. If the user input can\'t be parsed, you need to reply empty JSON []. '}, {'role': 'user', 'content': 'Give you some pictures e1.jpg, e2.png, e3.jpg, help me count the number of sheep?'}, {'role': 'assistant', 'content': '[{"task": "image-to-text", "id": 0, "dep": [-1], "args": {"image": "e1.jpg" }}, {"task": "object-detection", "id": 1, "dep": [-1], "args": {"image": "e1.jpg" }}, {"task": "visual-question-answering", "id": 2, "dep": [1], "args": {"image": "<GENERATED>-1", "text": "How many sheep in the picture"}} }}, {"task": "image-to-text", "id": 3, "dep": [-1], "args": {"image": "e2.png" }}, {"task": "object-detection", "id": 4, "dep": [-1], "args": {"image": "e2.png" }}, {"task": "visual-question-answering", "id": 5, "dep": [4], "args": {"image": "<GENERATED>-4", "text": "How many sheep in the picture"}} }}, {"task": "image-to-text", "id": 6, "dep": [-1], "args": {"image": "e3.jpg" }},  {"task": "object-detection", "id": 7, "dep": [-1], "args": {"image": "e3.jpg" }}, {"task": "visual-question-answering", "id": 8, "dep": [7], "args": {"image": "<GENERATED>-7", "text": "How many sheep in the picture"}}]'}, {'role': 'user', 'content': 'Look at /e.jpg, can you tell me how many objects in the picture? Give me a picture and video similar to this one.'}, {'role': 'assistant', 'content': '[{"task": "image-to-text", "id": 0, "dep": [-1], "args": {"image": "/e.jpg" }}, {"task": "object-detection", "id": 1, "dep": [-1], "args": {"image": "/e.jpg" }}, {"task": "visual-question-answering", "id": 2, "dep": [1], "args": {"image": "<GENERATED>-1", "text": "how many objects in the picture?" }}, {"task": "text-to-image", "id": 3, "dep": [0], "args": {"text": "<GENERATED-0>" }}, {"task": "image-to-image", "id": 4, "dep": [-1], "args": {"image": "/e.jpg" }}, {"task": "text-to-video", "id": 5, "dep": [0], "args": {"text": "<GENERATED-0>" }}]'}, {'role': 'user', 'content': 'given a document /images/e.jpeg, answer me what is the student amount? And describe the image with your voice'}, {'role': 'assistant', 'content': '{"task": "document-question-answering", "id": 0, "dep": [-1], "args": {"image": "/images/e.jpeg", "text": "what is the student amount?" }}, {"task": "visual-question-answering", "id": 1, "dep": [-1], "args": {"image": "/images/e.jpeg", "text": "what is the student amount?" }}, {"task": "image-to-text", "id": 2, "dep": [-1], "args": {"image": "/images/e.jpg" }}, {"task": "text-to-speech", "id": 3, "dep": [2], "args": {"text": "<GENERATED>-2" }}]'}, {'role': 'user', 'content': 'Given an image /example.jpg, first generate a hed image, then based on the hed image generate a new image where a girl is reading a book'}, {'role': 'assistant', 'content': '[{"task": "openpose-control", "id": 0, "dep": [-1], "args": {"image": "/example.jpg" }},  {"task": "openpose-text-to-image", "id": 1, "dep": [0], "args": {"text": "a girl is reading a book", "image": "<GENERATED>-0" }}]'}, {'role': 'user', 'content': "please show me a video and an image of (based on the text) 'a boy is running' and dub it"}, {'role': 'assistant', 'content': '[{"task": "text-to-video", "id": 0, "dep": [-1], "args": {"text": "a boy is running" }}, {"task": "text-to-speech", "id": 1, "dep": [-1], "args": {"text": "a boy is running" }}, {"task": "text-to-image", "id": 2, "dep": [-1], "args": {"text": "a boy is running" }}]'}, {'role': 'user', 'content': 'please show me a joke and an image of cat'}, {'role': 'assistant', 'content': '[{"task": "conversational", "id": 0, "dep": [-1], "args": {"text": "please show me a joke of cat" }}, {"task": "text-to-image", "id": 1, "dep": [-1], "args": {"text": "a photo of cat" }}]'}, {'role': 'user', 'content': 'The chat log [ [] ] may contain the resources I mentioned. Now I input { are you there? }. Pay attention to the input and output types of tasks and the dependencies between tasks.'}]
2023-04-06 12:17:37,629 - __main__ - DEBUG - {"id":"cmpl-72SD92uog8Bti5ia76mrn42cDwfD0","object":"text_completion","created":1680819455,"model":"text-davinci-003","choices":[{"text":"\n[{\"task\": \"conversational\", \"id\": 0, \"dep\": [-1], \"args\": {\"text\": \"are you there?\" }}]","index":0,"logprobs":null,"finish_reason":"stop"}],"usage":{"prompt_tokens":1906,"completion_tokens":34,"total_tokens":1940}}
2023-04-06 12:17:37,629 - __main__ - INFO - [{"task": "conversational", "id": 0, "dep": [-1], "args": {"text": "are you there?" }}]
2023-04-06 12:17:37,629 - __main__ - DEBUG - [{'task': 'conversational', 'id': 0, 'dep': [-1], 'args': {'text': 'are you there?'}}]
2023-04-06 12:17:37,649 - __main__ - DEBUG - Run task: 0 - conversational
2023-04-06 12:17:37,649 - __main__ - DEBUG - Deps: []
2023-04-06 12:17:37,649 - __main__ - DEBUG - parsed task: {'task': 'conversational', 'id': 0, 'dep': [-1], 'args': {'text': 'are you there?'}}
2023-04-06 12:17:39,131 - __main__ - DEBUG - {"id":"cmpl-72SDCqpG1E3CmxfKirBtFkdui9fIJ","object":"text_completion","created":1680819458,"model":"text-davinci-003","choices":[{"text":"\nYes, I am here. What can I help you with?","index":0,"logprobs":null,"finish_reason":"stop"}],"usage":{"prompt_tokens":112,"completion_tokens":14,"total_tokens":126}}
2023-04-06 12:17:39,132 - __main__ - DEBUG - inference result: {'response': 'Yes, I am here. What can I help you with?'}
2023-04-06 12:17:39,145 - __main__ - DEBUG - {0: {'task': {'task': 'conversational', 'id': 0, 'dep': [-1], 'args': {'text': 'are you there?'}}, 'inference result': {'response': 'Yes, I am here. What can I help you with?'}, 'choose model result': {'id': 'ChatGPT', 'reason': 'ChatGPT is the best model for this task.'}}}
2023-04-06 12:17:39,145 - __main__ - DEBUG - [{'role': 'system', 'content': '#4 Response Generation Stage: With the task execution logs, the AI assistant needs to describe the process and inference results.'}, {'role': 'user', 'content': 'are you there?'}, {'role': 'assistant', 'content': "Before give you a response, I want to introduce my workflow for your request, which is shown in the following JSON data: [{'task': {'task': 'conversational', 'id': 0, 'dep': [-1], 'args': {'text': 'are you there?'}}, 'inference result': {'response': 'Yes, I am here. What can I help you with?'}, 'choose model result': {'id': 'ChatGPT', 'reason': 'ChatGPT is the best model for this task.'}}]. Do you have any demands regarding my response?"}, {'role': 'user', 'content': "Yes. Please first think carefully and directly answer my request based on the inference results. Then please detail your workflow step by step including the used models and inference results for my request in your friendly tone. Please filter out information that is not relevant to my request. If any generated files of images, audios or videos in the inference results, must tell me the complete path. If there is nothing in the results, please tell me you can't make it. }"}]
2023-04-06 12:17:45,152 - __main__ - DEBUG - {"id":"cmpl-72SDDKUwzjd8JCFhpvEvn8JEN3JN0","object":"text_completion","created":1680819459,"model":"text-davinci-003","choices":[{"text":"\nSure. I understand your request. Based on the inference results, my response to your request is \"Yes, I am here. What can I help you with?\". My workflow for this request is as follows: First, I received your request \"are you there?\" and identified it as a conversational task. Then, I chose the ChatGPT model as the best model for this task. Finally, I used the ChatGPT model to generate the response \"Yes, I am here. What can I help you with?\". There are no generated files of images, audios or videos in the inference results. Is there anything else I can help you with?","index":0,"logprobs":null,"finish_reason":"stop"}],"usage":{"prompt_tokens":296,"completion_tokens":134,"total_tokens":430}}
2023-04-06 12:17:45,153 - __main__ - INFO - response: Sure. I understand your request. Based on the inference results, my response to your request is "Yes, I am here. What can I help you with?". My workflow for this request is as follows: First, I received your request "are you there?" and identified it as a conversational task. Then, I chose the ChatGPT model as the best model for this task. Finally, I used the ChatGPT model to generate the response "Yes, I am here. What can I help you with?". There are no generated files of images, audios or videos in the inference results. Is there anything else I can help you with?
```

**Comment by HaroldMitts** [comment]
Somehow its now working. What I did was double click on the settings icon in the webpage. This exposed my api key into the text box, then I pressed save. It still did not work, but next, I noticed now the drop down menu had more options besides the default value. Now it contained terminal, polish, translation, and default. I chose translation and entered a simple phrase then submit. it translated successfully. Next, I switch back to default and it now works magically...

![image](https://user-images.githubusercontent.com/13913671/230512900-b6053787-d7fe-4007-847e-4aa108c70f0f.png)
